pkg/cloudprovider/providers/gce/gce.go

```diff
@@ -852,6 +856,15 @@ func (gce *GCECloud) createTargetPool(name, serviceName, region string, hosts []
 	for _, host := range hosts {
 		instances = append(instances, makeHostURL(gce.projectID, host.Zone, host.Name))
 	}
+	if len(instances) > maxInstancesPerTargetPool {
```

Comment:
We don't have the list of Nodes here, which means pulling another O(thousands) list.  It's probably the right thing to do, but I feel like this is moot eventually and we don't filter for healthy nodes today...